sample/Kmd.Logic.Identity.Authorization.Sample/Program.cs

```diff
@@ -0,0 +1,135 @@
+            catch (Exception ex)
+            {
+                Log.Fatal(ex, "Caught a fatal unhandled exception");
+            }
+#pragma warning restore CA1031 // Do not catch general exception types
+            finally
+            {
+                Log.CloseAndFlush();
+            }
+        }
+
+        private static void InitLogger()
+        {
+            Log.Logger = new LoggerConfiguration()
+                .Enrich.FromLogContext()
+                .Destructure.ByTransforming<JsonWebToken>(jwt =>
+                {
+                    var result = new Dictionary<string, object>();
+
+                    foreach (var claim in jwt.Claims)
+                    {
+                        if (result.TryGetValue(claim.Type, out var value))
+                        {
+                            var list = value as List<string>;
+
+                            if (list == null)
+                            {
+                                list = new List<string>();
+                                list.Add((string)value);
+                                result[claim.Type] = list;
+                            }
+
+                            list.Add(claim.Value);
+                        }
+                        else
+                        {
+                            var claimValue = claim.Value;
+
+                            switch (claim.Type)
+                            {
+                                case "exp":
+                                case "nbf":
+                                case "iat":
+                                    if (long.TryParse(claimValue, out var timestamp))
+                                    {
+                                        var dt = DateTimeOffset.FromUnixTimeSeconds(timestamp).ToLocalTime();
+                                        claimValue += $" (which is {dt})";
+                                    }
+
+                                    break;
+                            }
+
+                            result.Add(claim.Type, claimValue);
+                        }
+                    }
+
+                    return result;
+                })
+                .WriteTo.Console()
+                .CreateLogger();
+        }
+
+        private static async Task Run(AppSettings configuration)
+        {
+            var validator = new ConfigurationValidator(configuration);
+            if (!validator.Validate())
+            {
+                return;
+            }
+
+            var options = new LogicTokenProviderOptions
+            {
+                ClientId = configuration.ClientId,
+                ClientSecret = configuration.ClientSecret,
+            };
+
+            if (!string.IsNullOrEmpty(configuration.AuthorizationScope))
+            {
+                options.AuthorizationScope = configuration.AuthorizationScope;
+            }
+
+            if (configuration.AuthorizationTokenIssuer != null)
```

Comment:
This will always be null unless the user modifies the appsettings right?

Reply:
Yes